springbackend/src/main/java/com/simplecoding/social/controller/PrivateEndpoint.java

```diff
@@ -28,4 +44,22 @@
         return ResponseEntity.ok(userDto);
     }
 
+    /*@GetMapping("addFriend")
+    public ResponseEntity<?> addUserAsFriend(@AuthenticationPrincipal UserDto userDto){
+        User user = userRepository.getOne(2);
+        UserDto userDto2 = modelMapper.map(user,UserDto.class);
+        friendService.saveFriend(userDto,userDto2);
+        return ResponseEntity.ok("ok");
+    }*/
+
+    @GetMapping("addFriend")
+    public ResponseEntity<?> addUser(@AuthenticationPrincipal @RequestParam("id")String id){
+        UserDto currentUser = securityService.getUser();
+        User user = userRepository.getOne(Integer.parseInt(id));
+        UserDto userDto2 = modelMapper.map(user,UserDto.class);
+        friendService.saveFriend(currentUser,userDto2);
+        friendService.saveFriend(userDto2,currentUser);
```

Comment:
change param to addUser(@RequestParam("friend_Id")String friendId)

refactor the code, friendService.saveFriend(currentUser, friendId ); and move all logic to friendService,